packages/devtools_app/lib/src/utils.dart

```diff
@@ -443,3 +443,42 @@ double safeDivide(num numerator, num denominator, {double ifNotFinite = 0.0}) {
   }
   return ifNotFinite;
 }
+
+/// A simple change notifier.
```

Comment:
We should use [ChangeNotifier] or [ValueListenable] rather than writing our own similar class.
https://api.flutter.dev/flutter/foundation/ChangeNotifier-class.html
https://api.flutter.dev/flutter/foundation/ValueListenable-class.html

Reply:
The ChangeNotifier interface has some quirks to it that can make it cumbersome for passing the state changes that we want to use.

1. Registering a callback directly:
The following is not possible with a ChangeNotifier:

```dart
notifier.listen(() {
  setState(() {
    doSomething();
  });
});
```
This is because there's no way to pass that callback in to removeListener.

2. ChangeNotifier is O(N²) for dispatching notifications.

3. It's possible to accidentally add the same listener multiple times to ChangeNotifier.

By using a map instead of a list in the implementation, we can avoid all of these drawbacks.

I think that's worth the 12 extra lines of code.